### DISCUSSION TITLE
Wird self.STATE in Entity benötigt?

### DISCUSSION THREAD

**Discussion by SukramJ** [SukramJ/aiohomematic#14] [General]
Warum wird diese Code-Stelle benötigt?
https://github.com/SukramJ/hahomematic/blob/034ea62c270436db72746b5fa95f073ae5be42a7/hahomematic/entity.py#L67

Hierdurch wird die Initialisierung sehr langsam, da für jede Entität nochmal ein Request zur CCU aufgemacht wird, um aktuelle Daten zu holen.

**Comment by danielperna84** [comment]
Die Idee hierbei war den aktuellen Status eines Parameters zu erfassen. Hintergrund ist, dass manche Geräte (vor allem batteriebetriebene Sensoren) ihren Zustand nur selten bis gar nicht verschicken. Wenn man den letzten Zustand also nicht in irgendeinem Cache zwischenspeichert (was man definitiv tun sollte), dann wäre der Zustand in HA die ganze Zeit unbekannt.

Natürlich ist es beim Boot viel sinnvoller den aktuellen Zustand nicht abzufragen, weil das bei vielen Parametern ewig dauert. Perspektivisch soll das also auf jeden Fall weg. Es wird dann allerdings eine Alternative gebraucht, welche in irgendeiner Form den aktuellen Zustand abfragt. Z.B.  nur für Parameter die noch auf _UNKNOWN_ stehen nachdem ein Timeout abgelaufen ist.

**Reply by SukramJ** [reply]
Aktuell würde ich `self.STATE` entfernen, da ansonsten die Komponente nicht schnell genug startet. Ich glaube HA setzt eine Warnung nach 10 Sekunden und bricht nach 60 Sekunden ab. Ich kann mich da aber Irren. 
Eine nachträglich Abfrage bei Status unknown finde ich die bessere Alternative.

**Reply by danielperna84** [reply]
Hier noch die Idee: bei der verzögerten Abfrage könnte man noch unterscheiden zwischen Geräten die mit Batterie laufen (die haben ja für gewöhnlich einen Parameter für die Ladung) und denen die direkt am Strom hängen. Kann mir vorstellen, dass die die am Strom hängen zuverlässig zyklisch den Status verschicken. Bzw., bei denen würde es nichts _kosten_ dort einen kurzen Zyklus einzustellen (wäre dann etwas, was in der Doku stehen sollte). Dann müsste man am Ende idealerweise nur noch für die übrigen mit Batterie checken was der Zustand ist.

**Reply by SukramJ** [reply]
Umsetzung:
Die in HA **verwendeten** Entitäten laden bei der Initialisierung einmalig den State aus der CCU. Danach kommen die State Änderungen ausschließlich aus den von der CCU gesendeten Events.

**Reply by ThomDietrich** [reply]
Um generell bessere Klarheit zu verschaffen würde ich das nicht als Property implementieren. Statt `STATE`, was eine Konstante repräsentieren sollte wäre `update_state()` deutlich intuitiver.

Seitennotiz: Schon mal über Vergrößerung auf 120 oder 160 Zeichen nachgedacht? Die verlinkte Datei könnte deutlich übersichtlicher sein.

Edit: Entschuldigt, ich habe da wohl auf eine völlig veraltete Diskussion reagiert.
